test/SILGen/convenience_init_peer_delegation.swift

```diff
@@ -0,0 +1,85 @@
+  convenience init(convenience: ()) {
+    self.init()
+  }
+
+  // ...but can statically invoke peer convenience inits
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC17doubleConvenienceACyt_tcfC
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC17doubleConvenienceACyt_tcfc
+  // CHECK:         function_ref @$S32convenience_init_peer_delegation1XC0A0ACyt_tcfc
+  convenience init(doubleConvenience: ()) {
+    self.init(convenience: ())
+  }
+
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC8requiredACyt_tcfC
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC8requiredACyt_tcfc
+  required init(required: ()) {
+  }
+
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC19requiredConvenienceACyt_tcfC
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC19requiredConvenienceACyt_tcfc
+  required convenience init(requiredConvenience: ()) {
+    self.init(required: ())
+  }
+
+  // Convenience inits must dynamically dispatch required peer convenience inits
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC25requiredDoubleConvenienceACyt_tcfC
+  // CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation1XC25requiredDoubleConvenienceACyt_tcfc
+  // CHECK:         class_method %7 : $X, #X.init!initializer.1
+  required convenience init(requiredDoubleConvenience: ()) {
+    self.init(requiredDoubleConvenience: ())
+  }
+}
+
+// CHECK-LABEL: sil hidden @$S32convenience_init_peer_delegation11invocations2xtyAA1XCm_tF
+func invocations(xt: X.Type) {
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XCACycfC
+  _ = X()
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XC0A0ACyt_tcfC
+  _ = X(convenience: ())
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XC17doubleConvenienceACyt_tcfC
+  _ = X(doubleConvenience: ())
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XC8requiredACyt_tcfC
+  _ = X(required: ())
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XC19requiredConvenienceACyt_tcfC
+  _ = X(requiredConvenience: ())
+  // CHECK: function_ref @$S32convenience_init_peer_delegation1XC25requiredDoubleConvenienceACyt_tcfC
+  _ = X(requiredDoubleConvenience: ())
+
+  // CHECK: class_method %0 : $@thick X.Type, #X.init!allocator.1
+  _ = xt.init(required: ())
+  // CHECK: class_method %0 : $@thick X.Type, #X.init!allocator.1
+  _ = xt.init(requiredConvenience: ())
+  // CHECK: class_method %0 : $@thick X.Type, #X.init!allocator.1
+  _ = xt.init(requiredDoubleConvenience: ())
+}
+
+// CHECK-LABEL: sil_vtable X
+//                -- designated init()
+// CHECK-NOT:     @$S32convenience_init_peer_delegation1XCACycfC
+// CHECK:         @$S32convenience_init_peer_delegation1XCACycfc
+
+//                -- no unrequired convenience inits
+// CHECK-NOT:     @$S32convenience_init_peer_delegation1XC0A0ACyt_tcfC
```

Comment:
CHECK-NOT feels fragile because if the mangling ever changes slightly, *and* the vtable entry re-appears, the test won't catch it. How about SILVerifier checks instead that non-required constructors don't show up in the vtable?

Reply:
Sure, would it be safe to do a more general verifier check that SILVTables only include entries visited by the `SILVTableVisitor`?

Reply:
Yes, an entry should appear in the vtable iff its visited by the vtable visitor, but you also want the vtable visitor to only visit the right decls...